src/main/shell-sync.ts

```diff
@@ -11,9 +11,10 @@ interface Env {
  * useful on macos where this always needs to be done.
  * @param locale Should be electron's `app.getLocale()`
  */
-export function shellSync(locale: string) {
+export async function shellSync(locale: string) {
   const { shell } = os.userInfo();
-  const env: Env = JSON.parse(JSON.stringify(shellEnv.sync(shell)))
+  
+  const env: Env = JSON.parse(JSON.stringify(await shellEnv(shell)))
```

Comment:
btw, we have `common/utils/cloneJson.ts` for cloning plain object with JSON.stuff